src/java/com/puppetlabs/jruby_utils/pool/JRubyPool.java

```diff
@@ -86,6 +86,12 @@
     // to impose any noticeable performance degradation.
     private volatile Thread poolLockThread = null;
 
+    // Holds a poison pill object for errors and shutdowns
+    // If not null, takes priority over any pool instance when a call to
+    // borrowItem is made. Returns made using releaseItem are ignored if the
+    // released item is the poison pill already stored here
+    private E poisonPill;
```

Comment:
I think we would want to mark this variable as `volatile` like we do the `poolLockThread` above since the value can be gotten / set from different threads concurrently.  @KevinCorcoran - that sound right to you?

Cosmetic suggestion - maybe it would be good to just call this `pill` like we do with the existing `insertPill` method?  We use the term 'poison pill' informally but with the different kinds of pills that can actually be inserted into the pool, might be better to just go with the more generic name everywhere.